packages/web-components/src/select/select.ts

```diff
@@ -0,0 +1,206 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { ARIAGlobalStatesAndProperties } from '@microsoft/fast-foundation';
+import { SelectSize } from './select.options.js';
+
+/**
+ * The base class used for constructing a fluent-slider custom element
+ * @public
+ */
+export class Select extends FASTElement {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    Object.assign(this, new ARIAGlobalStatesAndProperties()); // Assigns ARIA global states and properties to the element
+  }
+
+  // Attributes
+  /**
+   * The name of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name?: string;
+
+  /**
+   * The required boolean attribute of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: required
+   */
+  @attr({ mode: 'boolean' })
+  public required?: boolean;
+
+  /**
+   * The optional label of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: label
+   */
+  @attr
+  public label?: string;
+
+  /**
+   * The autofocus attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autofocus
+   */
+  @attr({ attribute: 'autofocus', mode: 'boolean' })
+  public autofocusAttr?: boolean;
+
+  /**
+   * The autocomplete attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autocomplete
+   */
+  @attr
+  public autocomplete?: string;
+
+  /**
+   * The size of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: control-size
+   */
+  @attr
+  public controlSize?: SelectSize;
```

Comment:
Given there isn't multiple, can this just be size?

Reply:
I had thought about that, but after our conversation with @miroslavstastny on Monday I was under the impression that we wanted similar attributes that controlled how large and small a control should be–especially for similar elements like Select, Dropdown, Combobox.

Reply:
@procload I'll let @miroslavstastny chime in here, but my understanding was, *when* we can have an aligned API where there is no conflict, we should. So in the case of this, the visual size *can* be size still. The outliers we will likely need to document are dropdown and possibly combobox.